Add PyPy support for Mac OSX builds

HexDecimal opened the pull request:
I've made progress on getting PyPy to make wheels on Mac OSX.  I wanted to show off what I had so far.

I tried to get Linux working too, but auditwheel didn't handle PyPy wheels as well as delocate does.

Here is a PYPY_VERSION=5.4 job running correctly on one of my projects:
https://travis-ci.org/HexDecimal/libtcod-cffi-wheels/jobs/165707608

What's left:  I think I might still need to work on the documentation of the code.  I haven't added anything to the README yet.  There are 3 environment variables that select which Python to install but it just picks the first one it sees right now when maybe it should raise an error or something.

There's a little more information in the commit log.

There also seems to be in issue with Travis's current ruby install causing `shell_session_update: command not found` errors.  My example job worked because of the added command: `rvm get stable --auto-dotfiles`

matthew-brett commented:
Sorry to be (terribly) slow to get back to you.   Are you still interested in working on this?

HexDecimal commented:
I've fixed the current conflicts.  I'm also dropping PyPy3 until it becomes stable.

I've added a PyPy environment to the matrix in the readme, but I haven't tested it on a real repository after the rebase.  I assume it still works since the TravisCI tests are passing.

matthew-brett commented:
Can you think if a nice way of specifying pypy within the `MB_PYTHON_VERSION` ?

HexDecimal commented:
Done.

You now select the PyPy version like this: `MB_PYTHON_VERSION=pypy-5.7`

matthew-brett commented on osx_utils.sh:
```diff
@@ -219,10 +248,9 @@ function get_macpython_environment {
+    local venv_dir=$1
     remove_travis_ve_pip
-    install_macpython $version
+    install_python
```
Have personal, but not overriding preference for passing value of MB_PYTHON_VERSION to function, to make the function easier to test, and more transparent (in the sense that you don't need know the global variable state).  Likewise for the get_python_environment etc.

HexDecimal commented:
I removed the MB_PYTHON_VERSION variable from `osx_utils.sh`.  Now you can pass a PyPy implementation to the functions that used to only take a CPython version number such as `get_macpython_environment`, and `install_macpython`.  This is much better than what I was originally doing.

matthew-brett commented:
Excellent - thanks.  One last request - can you add some tests for `fill_pypy_ver` (see `test_fill_pyver.sh`) and `get_pypy_build_prefix`.

HexDecimal commented:
I think the `get_pypy_build_prefix` function is already covered by the current Travis matrix.

`fill_pypy_ver` was made to be easy to work with and update,  and I'm very confident in the `unroll_version` function.  My changes to `test_python_install.sh` already check that the requested version is a sub-string of the Python implementation actually installed.

I don't think these need dedicated tests. Travis should have failed if they didn't work.

matthew-brett commented:
Always a bold statement: 'doesn't need any more tests'...

I'll merge this one; if I have time, I'll add some tests.

Thanks for persisting.

matthew-brett commented:
See: https://github.com/matthew-brett/multibuild/pull/22

HexDecimal commented:
`PyPy<=2.1` doesn't have their archived folders matching their filenames.  `install_mac_pypy` is failing to set `PYTHON_EXE` correctly for those versions at the moment.
